bvm/ballerina-core/src/main/java/org/ballerinalang/helpers/DeepEqualHelper.java

```diff
@@ -0,0 +1,271 @@
+/*
+*  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+
+package org.ballerinalang.helpers;
+
+import org.ballerinalang.model.types.BStructType;
+import org.ballerinalang.model.types.TypeTags;
+import org.ballerinalang.model.util.JsonNode;
+import org.ballerinalang.model.values.BConnector;
+import org.ballerinalang.model.values.BEnumerator;
+import org.ballerinalang.model.values.BJSON;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BNewArray;
+import org.ballerinalang.model.values.BRefType;
+import org.ballerinalang.model.values.BStruct;
+import org.ballerinalang.model.values.BValue;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A helper class for the deep equals implementation.
+ */
+public class DeepEqualHelper {
+    /**
+     * Deep equal or strict equal implementation for any value type. I.E ===.
+     * @param lhVal The value on the left side.
+     * @param rhVal The value on the right side.
+     * @return True if values are equal, else false.
+     */
+    public static boolean isDeepEqual(BValue lhVal, BValue rhVal) {
```

Comment:
s/lhVal/lhsValue/g and s/rhVal/rhsValue

Consistent with the other parts of the BVM

Reply:
Renamed in https://github.com/ballerina-lang/ballerina/pull/4532/commits/de2bf293cdf607e12055f2e7e3bb4ab78276774d